src/lib/temple/front/ab-test.provider.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useMemo } from 'react';
+
+import constate from 'constate';
+
+import { useAnalyticsState } from 'lib/analytics/use-analytics-state.hook';
+import { useRetryableSWR } from 'lib/swr';
+import { getABGroup } from 'lib/templewallet-api';
+
+import { usePassiveStorage } from './storage';
+
+export function useAB() {
+  const { data: abGroup, loading } = useABGroup();
+  const { analyticsState } = useAnalyticsState();
+
+  const [localABGroup, setLocalABGroup] = usePassiveStorage<'A' | 'B' | null>('ab-test-value', null);
+
+  useEffect(() => {
+    if (!loading && !localABGroup) {
+      setLocalABGroup(abGroup);
+    }
+  }, [localABGroup, loading, setLocalABGroup, abGroup]);
+
+  return useMemo(() => {
+    if (analyticsState.enabled) {
+      return localABGroup;
+    }
+    return null;
+  }, [analyticsState, localABGroup]);
+}
+
+export const [ABTestGroupProvider, useABGroup] = constate((params: { suspense?: boolean }) => {
+  const { data, isValidating: loading } = useRetryableSWR('ab-test-group', fetchABGroup, {
+    revalidateOnMount: false,
+    dedupingInterval: 30_000,
```

Comment:
would this re-fetch a new value for each 30 secs?

Reply:
no, its just doing fetch every 30 seconds until request OK and then just revalidates it locally